hack/heap/hh_shared.c

```diff
@@ -592,7 +594,7 @@ void hh_add_dep(value ocaml_dep) {
   htable_add(deptbl, dep >> 31, dep);
```

Comment:
ah, finally figured out what was wrong in our internal tests (and I'm still not sure why it doesn't repro in travis).

adding `CAMLparam1` here without also adding `CAMLreturn0` here and the `return` a couple lines up meant that `ocaml_dep` was never freed and it'd OOM.

I'm including that fix, no need to update. should be landing soon. thanks for your patience!